synapse/storage/databases/main/devices.py

```diff
@@ -596,7 +602,7 @@ def _mark_as_sent_devices_by_remote_txn(
             txn=txn,
             table="device_lists_outbound_last_success",
             key_names=("destination", "user_id"),
-            key_values=((destination, user_id) for user_id, _ in rows),
+            key_values=tuple((destination, user_id) for user_id, _ in rows),
```

Comment:
Alternately I think we can make this a list?

```suggestion
            key_values=[destination, user_id) for user_id, _ in rows],
```

I don't think I have any preference, but I think we generally default to lists. 🤷

Reply:
I have a (very) minor preference for passing around immutable tuples, but I don't think it makes sense to do it in an expression directly passed to a function call.

Reply:
I have no strong preference. 👍